app/assets/javascripts/show_errors.js

```diff
@@ -2,51 +2,48 @@
 // For it to work properly, you need to pass data from the controller like:
 //   format.json { render json: {errors: link.errors, name: 'link'}, status: :unprocessable_entity }
 // The name field is for if the form element names are prefixed, eg 'link[title]'
+let ErrorDisplay = {
 
-// this is ripe for a refactor to use the event system, and could be a backbone view too
-
-window.Champaign = window.Champaign || {};
-window.Champaign.showErrors = function(e, data) {
-
-  if (!e || !data || !data.responseText || data.status != 422) {
-    return; // no reason to try if we dont have what we need
-  }
-
-  // use the relevant form if the event was a form submission
-  $form = ($(e.target) && $(e.target).length > 0) ? $(e.target) : $('form');
-  response = $.parseJSON(data.responseText);
-
-  var errorMsg = function(field_name, msgs) {
-    var msg = (typeof msgs === "string") ? msgs : msgs[0]
-    var prefix = window.I18n ? I18n.t('errors.this_field') : 'This field';
-    return ["<div class='error-msg'>", prefix, " ", msg, "</div>"].join('');
-  }
-
-  var clearErrors = function() {
-    $form.find('has-error').removeClass('has-error');
+  show(e, data) {
+    if (!e || !data || !data.responseText || data.status != 422) {
+      return; // no reason to try if we dont have what we need
+    }
+    // use the relevant form if the event was a form submission
+    this.$form = ($(e.target) && $(e.target).length > 0) ? $(e.target) : $('form');
+    this.response = $.parseJSON(data.responseText);
+    this.clearErrors(this.$form);
+    $.each(this.response.errors, (f, m) => { this.showError(f, m) });
+  },
+
+  clearErrors($form) {
```

Comment:
this doesn't actually get instantiated... which is a pretty weird design pattern. I will probably change it to stop using instance variables because it really shouldn't if it's all class methods. I'm hesitant though to have `clearErrors` not take an argument because it gets called by other js views that already have their own `this.$form` they can just pass right along.